### PR TITLE
Feat/str pad

### PR DESCRIPTION
**Pull request by ftamhar**
- Programming langugage: golang
- Function(s) that I'm implementing:[str_pad](https://www.php.net/manual/en/function.str-pad)
- Issue number: #36

- [x] I have created unit tests to support my implementation
- [x] I have created in-code documentation (or docstring) to document my code
- [x] (Optional) I have updated the README.md file to document the function I'm implementing

### CODE REVIEW

**Comment by ftamhar** [comment]
yok gas @WahidinAji @aldy505

File: go/str_pad_test.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			actual, err := StrPad(tt.s, tt.length, tt.args...)
+			assert.Equal(t, tt.expectedString, actual)
+			assert.Equal(t, tt.expectedError, err)
+
+			if tt.length > len(tt.s) && tt.expectedError == nil {
+				assert.Equal(t, tt.length, len(actual))
+			}
```
Comment (aldy505) [inline go/str_pad_test.go:290]: Please do not use testify/assert, don't add additional dependency.

File: go/str_pad.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	padString := " "
+
+	// set pad
+	if len(args) > 0 {
+		switch args[0].(type) {
+		case string, float64, int:
+			padString = fmt.Sprintf("%v", args[0])
+		default:
+			return "", ErrPadInvalid
+		}
```
Comment (aldy505) [inline go/str_pad.go:51]: `padString` can also be a `boolean` type. Try:
```php
<?php
echo str_pad("Alien", 12, true); // returns "Alien1111111"
```

File: go/str_pad.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	var padType int
+	if len(args) == 2 {
+		switch args[1].(type) {
+		case int:
+			padType = args[1].(int)
+		case float64:
+			padType = int(args[1].(float64))
+		default:
+			return "", ErrPadTypeInvalid
+		}
```
Comment (aldy505) [inline go/str_pad.go:69]: According to the `str_pad` [documentation here](https://www.php.net/manual/en/function.str-pad.php), `padType` can also be a string. Try:
```php
<?php
echo str_pad("Alien", 12, " ", "1");
```

**Review by aldy505 — CHANGES_REQUESTED** [review]

File: go/str_pad.go (hunk 2 above)
Comment (ftamhar) [reply go/str_pad.go:51]: di kasih false malah error, gila ini php wkwkwkw

**Review by aldy505 — APPROVED** [review]

**Pull request merged**